### PR TITLE
Add a banner that displays when a deposit is sent for review

### LINKED ISSUE #461: Banner when approval needed
If form is filled out properly, and the collection uses the review workflow, then the user returns to deposit view page with a banner message that says 
“Your deposit has been sent for approval. You will receive an email once your deposit has been approved.”

### PR DESCRIPTION
**Pull request by jcoyne**
## Why was this change made?
Fixes #461

### CODE REVIEW

File: app/components/works/detail_component.rb
Comment (peetucket) [inline app/components/works/detail_component.rb:70]: i'm unfamiliar with `helpers.allowed_to?(XXXX)` ... where is that defined/coming from?

**Review by peetucket — APPROVED** [review]

**Comment by peetucket** [comment]
Looks like a rebase is needed now

File: app/components/works/detail_component.rb
Comment (mjgiarlo) [reply app/components/works/detail_component.rb:70]: `helpers` is a container object that is available to all view components, and `allowed_to?` is a helper method set up by ActionPolicy.

Comment (peetucket) [reply app/components/works/detail_component.rb:70]: ok, so `app/policies/work_policy.rb` is being used (because we are asking about a `work`) and we are hitting the `review?` method defined there

i'm familiar with the traditional rails helpers available to views, so there is an equivalent for components but stored elsewhere?  thanks

Comment (mjgiarlo) [reply app/components/works/detail_component.rb:70]: @peetucket :speech_balloon: 
> ok, so `app/policies/work_policy.rb` is being used (because we are asking about a `work`) and we are hitting the `review?` method defined there

Yep!
 
> i'm familiar with the traditional rails helpers available to views, so there is an equivalent for components but stored elsewhere? thanks

Yeah, they're the same helpers, and the ViewComponent library pulls them in via this `helpers` method which is available to all view components: https://github.com/github/view_component/blob/ec91c69a0552b90e5d6a891ebb5dd642ed44f2f8/lib/view_component/base.rb#L117-L121

Comment (peetucket) [reply app/components/works/detail_component.rb:70]: thanks for the explanations and sorry if missed these earlier

and it looks like actionpolicy (and the state machine used for works/collections) are gems and not core rails

Comment (mjgiarlo) [reply app/components/works/detail_component.rb:70]: Correct!

**Comment by mjgiarlo** [comment]
@jcoyne you wanna fix the conflict or back away from this or what?

**Comment by jcoyne** [comment]
@mjgiarlo I don't know if the changes you made recently to the status satisfy this requirement already.

**Comment by mjgiarlo** [comment]
@jcoyne ok, maybe we should discuss with @amyehodge

**Comment by amyehodge** [comment]
@mjgiarlo and @jcoyne I'm happy to help, but not sure what the question is.

**Review by mjgiarlo — APPROVED** [review]

**Comment by mjgiarlo** [comment]
@amyehodge no worries! @jcoyne and I just chatted about this and it's good to go. This adds a banner to the work show page to set users' expectations a bit better w/r/t works waiting on approval.

**Pull request merged**
